rtgui/profilestore.cc

```diff
@@ -277,13 +277,14 @@ const ProfileStoreEntry* ProfileStore::findEntryFromFullPathU(Glib::ustring path
     if (path == DEFPROFILE_INTERNAL) {
         return internalDefaultEntry;
     }
+    // consistently apply casefold() to make sure dot position is correct
+    const Glib::ustring::size_type lastdot = path.casefold().find_last_of ('.');
 
-    size_t lastdot = path.find_last_of ('.');
-
-    if (lastdot != Glib::ustring::npos && lastdot <= path.size() - 4 && !path.casefold().compare (lastdot, 4, paramFileExtension))
+    if ((lastdot != Glib::ustring::npos) && (lastdot <= path.casefold().size() - 4) && (!path.casefold().compare (lastdot, 4, paramFileExtension)))
```

Comment:
Instead of calling `path.casefold()` up to three times here, you could save it first, like
```c++
const Glib::ustring casefolded_path = path.casefold();
```
and use it later.

And those additional parentheses... Well, you should know my opinion by now: Precedences are laid out well at large for a reason. The only common place you normally need parentheses is when mixing logical and bitwise operators. :smiley: